Staircase.js

```diff
@@ -182,47 +206,57 @@ Staircase.prototype.choose = function(goodAns) {
   var ans = (goodAns)
     ? 'harder'
     : 'easier';
+  if(!goodAns)
+    stair.startingExplorationScale = 1; // First time we get easier we stop the inital exploration scaling.
   var wait = (stair.wait)
     ? 'wait'
     : 'noWait';
-  return this.tasks[ans][stair.operation][wait][stair.direction](stair);
+  if(stair.verbosity>0)
+    console.log("Staircase '"+stair.name+"' ("+stair.operation+"): making test "+ans+", "+wait+" mode.");
+  var out = this.tasks[ans][stair.operation][wait][stair.direction](stair);
+  return out;
 };
-Staircase.prototype.checkLimits = function() {
+Staircase.prototype.checkLimits = function(stair) {
+  this.checkErr.ARG('checkLimits', arguments, 1);
+  this.checkErr.UNDEFINED(this.stairs, stair);
+    if(this.stairs[stair].val.length<2)
+        return;
   // check limits
-    if (stair.val[stair.val.length-1]<stair.val[stair.val.length-2] &&
-        stair.val[stair.val.length-1]<stair.limits[0]) {
-      stair.val[stair.val.length-1] = stair.limits[0];
-      stair.limitReached = true;
-    } else if (stair.val[stair.val.length-1]>stair.val[stair.val.length-2] &&
-        stair.val[stair.val.length-1]>stair.limits[1]) {
-      stair.val[stair.val.length-1] = stair.limits[1];
-      stair.limitReached = true;
+    if (this.stairs[stair].val[this.stairs[stair].val.length-1]<this.stairs[stair].val[this.stairs[stair].val.length-2] &&
+        this.stairs[stair].val[this.stairs[stair].val.length-1]<this.stairs[stair].limits[0]) {
+      this.stairs[stair].val[this.stairs[stair].val.length-1] = this.stairs[stair].limits[0];
+      this.stairs[stair].limitReached = true;
+    } else if (this.stairs[stair].val[this.stairs[stair].val.length-1]>this.stairs[stair].val[this.stairs[stair].val.length-2] &&
+        this.stairs[stair].val[this.stairs[stair].val.length-1]>this.stairs[stair].limits[1]) {
+      this.stairs[stair].val[this.stairs[stair].val.length-1] = this.stairs[stair].limits[1];
+      this.stairs[stair].limitReached = true;
     } else {
-      stair.limitReached = false;
+      this.stairs[stair].limitReached = false;
```

Comment:
Yes there was clearly a problem here. In the version I didn't push I came up with

```js
Staircase.prototype.checkLimits = function (currentStair) {
  var stair = currentStair;
  // check limits
  if (stair.val[stair.val.length - 1] < stair.val[stair.val.length - 2]
    && stair.val[stair.val.length - 1] < stair.limits[0]) {
    stair.val[stair.val.length - 1] = stair.limits[0];
    stair.limitReached = true;
  } else if (stair.val[stair.val.length - 1] > stair.val[stair.val.length - 2]
    && stair.val[stair.val.length - 1] > stair.limits[1]) {
    stair.val[stair.val.length - 1] = stair.limits[1];
    stair.limitReached = true;
  } else {
    stair.limitReached = false;
  }
};
```

Reply:
I think those are functionally the same, though yours is cleaner :)